tests/bgp/test_reliable_tsa.py

```diff
@@ -0,0 +1,1834 @@
+            clear_failed_flag_and_restart(duthost, service_name, container_name)
+        else:
+            pytest.fail("Failed to restart container '{}'".format(container_name))
+
+    logger.info("Container '{}' was restarted".format(container_name))
+
+
+def is_container_running(duthost, container_name):
+    """
+    @summary: Decide whether the container is running or not
+    @return:  Boolean value. True represents the container is running
+    """
+    result = duthost.shell("docker inspect -f \{{\{{.State.Running\}}\}} {}".format(container_name))  # noqa: W605
+    return result["stdout_lines"][0].strip() == "true"
+
+
+def check_container_state(duthost, container_name, should_be_running):
+    """
+    @summary: Determine whether a container is in the expected state (running/not running)
+    """
+    is_running = is_container_running(duthost, container_name)
+    return is_running == should_be_running
+
+
+def is_hiting_start_limit(duthost, service_name):
+    """
+    @summary: Determine whether the service can not be restarted is due to
+              start-limit-hit or not
+    """
+    service_status = duthost.shell("sudo systemctl status {}.service | grep 'Active'".format(service_name))
+    for line in service_status["stdout_lines"]:
+        if "start-limit-hit" in line:
+            return True
+
+    return False
+
+
+def clear_failed_flag_and_restart(duthost, service_name, container_name):
+    """
+    @summary: If a container hits the restart limitation, then we clear the failed flag and
+              restart it.
+    """
+    logger.info("{} hits start limit and clear reset-failed flag".format(service_name))
+    duthost.shell("sudo systemctl reset-failed {}.service".format(service_name))
+    duthost.shell("sudo systemctl start {}.service".format(service_name))
+    restarted = wait_until(300, 1, 0, check_container_state, duthost, container_name, True)
+    pytest_assert(restarted, "Failed to restart container '{}' after reset-failed was cleared".format(container_name))
+
+
+@pytest.mark.disable_loganalyzer
+def test_sup_tsa_act_when_sup_duts_on_tsb_initially(duthosts, localhost, enum_supervisor_dut_hostname,
+                                                    nbrhosts, traffic_shift_community, tbinfo):
+    """
+    Test supervisor TSA action when supervisor and line cards are in TSB initially
+    Verify supervisor config state changes to TSA and Line card BGP TSA operational state changes to TSA from TSB
+    Make sure only loopback routes are advertised to neighbors during line cards' TSA state.
+    """
+    suphost = duthosts[enum_supervisor_dut_hostname]
+    orig_v4_routes, orig_v6_routes = dict(), dict()
+    dut_nbrhosts = dict()
+    for linecard in duthosts.frontend_nodes:
+        dut_nbrhosts[linecard] = nbrhosts_to_dut(linecard, nbrhosts)
+        if not check_tsa_persistence_support(linecard):
+            pytest.skip("TSA persistence not supported in the image")
```

Comment:
TSA persistence support on the linecard exists on all images that supports T2 and since this is a T2-only test, this check is not needed. 

Should we be checking chassis-app-db for tsa_enabled field instead and skip if reliable tsa is not in the image?